numpy/lib/tests/test_arraypad.py

```diff
@@ -1260,3 +1260,38 @@ def test_check_wrong_pad_amount(self):
         kwargs = dict(mode='mean', stat_length=(3, ))
         assert_raises(TypeError, pad, arr, ((2, 3, 4), (3, 2)),
                       **kwargs)
+
+
+@pytest.mark.xfail(reason="not yet implemented, see gh-11358")
+def test_unsupported_mode():
+    """Test error message for unsupported pad modes."""
+    with pytest.raises(ValueError, message="mode '.*' is not supported"):
+        pad([1, 2, 3], 1, mode="unknown")
+    with pytest.raises(ValueError, message="mode '.*' is not supported"):
+        pad([1, 2, 3], 1, mode=3)
+    with pytest.raises(ValueError, message="mode '.*' is not supported"):
+        pad([1, 2, 3], 1, mode=None)
+
+
+@pytest.mark.xfail(reason="not yet implemented, see gh-11358")
+def test_order():
```

Comment:
I am good with putting this in as is (leaving new tests for later), if you like. One thing I would like here is a test for a case that is not contiguous (3d or 2d slices). And also a test for a 1D array, because it is both F and C-order.

Reply:
That test would consist of checking that `np.argsort(padded.strides) == np.argsort(orig.strides)`, for cases where the strides are not pre-sorted.

Reply:
I doubt we will add that in the first iteration, nor do I think it is a serious bug to return C-order, so I am not sure we need an xfail test for it (which would stay xfail indefinitely).